api/db-cdr/changelog/db.changelog-10.xml

```diff
@@ -0,0 +1,150 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.0.xsd
+        http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd">
+
+
+  <changeSet author="biopete" id="changelog-10.0" runOnChange="true">
+    <validCheckSum>ANY</validCheckSum>
+    <preConditions onFail="MARK_RAN">
+      <not>
+        <tableExists tableName="concept" schemaName="cdr"></tableExists>
```

Comment:
This would seem to contradict runOnChange="true". If you change the changeset, the concept table will already exist and it will try to run this again, and fail.

So we should either drop this precondition, or drop runOnChange="true"; or possibly both. Probably given that this schema shouldn't change, you don't really need either.

Reply:
Yeah. I was trying to see how runOnChange might work. You are correct. I will get rid of runOnChange.  I like the precondition because these tables will take some doing to fill and I don't want them getting wiped out unintentionally.

Reply:
They aren't going to get wiped out. If you don't specify runOnChange or runAlways, each migration runs exactly once. If you *did* specify runOnChange or runAlways, they might run multiple times; but even then it wouldn't drop a table unless you asked it to, it would just update it.

So you don't need the preconditions, either.

Reply:
Are we going to remove runOnChange="true" and preConditions?